src/events.rs

```diff
@@ -141,15 +149,8 @@ impl<T: System> EventsDecoder<T> {
         let size = U::default().encode().len();
         // A segmenter decodes a type from an input stream (&mut &[u8]) and returns the serialized
         // type to the output stream (&mut Vec<u8>).
-        self.type_segmenters.insert(
-            name.to_string(),
-            Box::new(
-                |input: &mut &[u8], output: &mut Vec<u8>| -> Result<(), Error> {
-                    U::decode(input).map_err(Error::from)?.encode_to(output);
-                    Ok(())
-                },
-            ),
-        );
+        self.type_segmenters
+            .insert(name.to_string(), Box::new(PhantomData::<U>));
```

Comment:
Just one more thing, can we have a tuple struct which wraps `PhantomData` to make this more readable.

Reply:
Good. Just fixed it.